### DISCUSSION TITLE
Welcome to uncertainties Discussions!

### DISCUSSION THREAD

**Discussion by lebigot** [lmfit/uncertainties#133] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about (unless they have a place on [StackOverflow](http://stackoverflow.com)).
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below!

**Comment by ipcoder** [comment]
Hi.
I would like to ask your opinion about the place of uncertainty calculations in data analysis in general.
More specifically, don't you think that most of the data being processed is fundamentally associated with uncertainties from its very source (measurements, confidence, statistics, etc).

Even if currently its not an acceptable norm, correct and rigorous calculations in most cases require some form of uncertainty processing and may prove very beneficial if become a standard part of any workflow.

Considering computational and "logistical" challenges (API, compatibility, etc) associated with treating the uncertainties as a separate task, I think that it must be integrated into the core libraries, like `numpy`, may be similar to masked arrays.

Efforts to support of uncertainties in `astropy` may be viewed as a confirmation of this idea.

What do you think about that? Are you aware of similar views or discussions among contributors or developers of core libraries like `numpy`,  `pandas`, or others?

**Comment by adriesse** [comment]
I have been searching for a place where GUM users discuss GUM usage, but haven't found one.  Does anyone know of such a place?

**Comment by cvarennes1650** [comment]
Hello, 
I would like to use your uncertainties package with a pandas dataframe rather than a numpy array. I see that pandas is compatible with several numpy commands. Is it possible to apply the uncertainties propagation to pandas columns? Thanks!

**Comment by lebigot** [comment]
What do you need to do exactly?

What works already is to create a Pandas DataFrame with numbers with uncertainties (for example by initializing it through a NumPy array with uncertainties, as created for instance with `uncertainties.unumpy.uarray`).

Also, using, for example, `unumpy.sin()` on a column works (but produces a NumPy array and not a Pandas Series). I'm thinking that it would be better if `unumpy.sin()` returned a Pandas series, like `numpy.sin()` does. Is this what you had in mind?

**Reply by lebigot** [reply]
I created an issue for this: https://github.com/lebigot/uncertainties/issues/150.

I must say that the last time I used my own program was maybe 10 years ago, and that with (again) very young children at home I don't have much time to work on uncertainties, but maybe someone will pick the issue up before me. :)
